src/components/organisms/SfCarousel/SfCarousel.js

```diff
@@ -68,15 +69,41 @@ export default {
   mounted: function() {
     const glide = new Glide(this.$refs.glide, this.mergedOptions);
     glide.mount();
+
     glide.on("run.before", move => {
-      const { slidePerPage } = this.mergedOptions;
+      const { slidePerPage, rewind, type } = this.mergedOptions;
 
       if (!slidePerPage) return;
 
       const { perView } = glide.settings;
+      if (!perView > 1) return;
+
+      const size = this.$slots.default.filter(slot => slot.tag).length;
       const { direction } = move;
+      let page, newIndex;
 
-      move.steps = direction === "<" ? perView : -perView;
+      switch (direction) {
+        case ">":
+        case "<":
+          page = Math.ceil(glide.index / perView);
+          newIndex = page * perView + (direction === ">" ? perView : -perView);
+          if (newIndex >= size) {
+            if (type === "slider" && !rewind) {
+              newIndex = glide.index;
+            } else {
+              newIndex = 0;
+            }
+          } else if (newIndex < 0) {
+            if (type === "slider" && !rewind) {
+              newIndex = glide.index;
+            } else {
+              newIndex = size - perView + (size % perView > 0 ? 1 : 0);
+            }
+          }
```

Comment:
When we have elements left over (eg.: 7 elements, 3 per page, so 2 full pages + 1 extra element) I think those elements may appear on last page:
(1-3)->(4-6)->(5->7)->(1-3)...

```js
if (newIndex < 0 || newIndex + perView > size) {
  if (type === "slider" && !rewind) {
    newIndex = glide.index;
  } else {
    newIndex = size - perView;
  }
}
```

wdyt?

Reply:
then the pages would always have the same elements

Reply:
Yes, you have right, I tried implemented carousel illusion but it doesn't look great :)